crates/compiler/mono/src/ir.rs

```diff
@@ -9647,6 +9647,9 @@ where
                 lambda_set_id = lambda_set_id.next();
 
                 stack.push(layout_interner.get(lambda_set.runtime_representation()));
+
+                // TODO: figure out if we need to look at the other layouts
+                // stack.push(layout_interner.get(lambda_set.ret));
```

Comment:
I'm guessing this TODO is for a future PR? (Fine if so, just wanted to double-check.)

Reply:
yes, it's something I noticed causing trouble when you have a signature like `{} -> (I64 -> I64)`. We never look at the return type when we hit a lambda set, and so no id is generated for the returned function.

but it's not relevant for this PR